src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilationBuilder.cs

```diff
@@ -192,7 +193,7 @@ public override ICompilation ToCompilation()
             if (_ibcTuning)
                 corJitFlags.Add(CorJitFlag.CORJIT_FLAG_BBINSTR);
 
-            JitConfigProvider.Initialize(corJitFlags, _ryujitOptions, _jitPath);
+            JitConfigProvider.Initialize(_context.Target.OperatingSystem, _context.Target.Architecture, corJitFlags, _ryujitOptions, _jitPath);
```

Comment:
It may be better to initialize JitConfigProvider with `_context.Target`, and let it pick the details it needs.

Reply:
Good point, changed in 23rd commit.